### ISSUE TITLE
insert or update on table "stock_move" violates foreign key constraint "stock_move_rule_id_fkey

### ISSUE THREAD

**Issue by niyshah** [OCA/OpenUpgrade#2870]
Upgrading Odoo CE 11 to Odoo CE 12

I receive the following error.

```
ERROR migration_v12 odoo.sql_db: bad query: ALTER TABLE "stock_move" ADD FOREIGN KEY ("rule_id") REFERENCES "stock_rule"("id") ON DELETE restrict
ERROR: insert or update on table "stock_move" violates foreign key constraint "stock_move_rule_id_fkey"
DETAIL:  Key (rule_id)=(1) is not present in table "stock_rule".

```
The table reference by **stock_move** in version 11 is not **stock_rule**, but **procurement_rule** (FOREIGN KEY (rule_id) REFERENCES procurement_rule(id) ON DELETE RESTRICT). **stock_rule** doesn't exist in version 11

However in V12 there is a stock_rule table (and no procurement_rule table).

As far as I can see I don't have any external modules installed that might cause this, nor can I see any corruption in the data.
I'm kind of expecting the migration tool to move/rename/copy the procurement_rule table data into the product_rule data.

I'm using the latest OpenUpgrade code (as of 28/7/2021)
I'm using the following command to run the upgrade on a copied database
`sudo -u odoo ./odoo-bin --config=/opt/odoo/.odoorc --update all --stop-after-init 

Can you provide any insight to aid my understanding in trying to solve this.
Thanks
Eric.

**Comment by niyshah** [comment]
Interesting find. I notice in the Live 11 database, that there are two legacy tables. 

openupgrade_legacy_11_0_procurement_rule
and
openupgrade_legacy_11_0_stock_location_path

It led me to wonder if the Upgrade from V10 didn't properly complete. 
I renamed the table procurement_rule to stock_rule and the stock module updated okay
Comments welcome

**Comment by MiquelRForgeFlow** [comment]
I don't know the cause of your problem. The rename is properly done in https://github.com/OCA/OpenUpgrade/blob/12.0/addons/stock/migrations/12.0.1.1/pre-migration.py#L58. Do you have openupgradelib to latest version?

**Comment by flotho** [comment]
ping @niyshah any more infos on this issue ?
personally, migration succeed from 10 -> 12.0

**Comment by alfonsosiloniz** [comment]
This problems happens when odoo-bin is being executed from OpenUpgrade but its addons are not taken in account. 

I was facing exactly this same problem, and after days working on it I realized that the OpenUpgrade modules where not taken in the start of odoo-bin, so it is exactly like having a Odoo 12 vanilla.

I needed to add this to the odoo-bin execution

./odoo-bin -d ddbb_v11_v12 -c /etc/odoo/odoo.conf --update all --stop-after-init  **--addons-path=./addons/**

Just adding this and the migration was clean and without any issue.

Hope it helps anybody looking for a solution
